keps/sig-node/4727-reasonable-image-gc/README.md

```diff
@@ -0,0 +1,320 @@
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Add an feature gate ImageGCBeforeStorageEviction, which denote image gc must occur before kubelet evict.
+When ImageGCBeforeStorageEviction is false or is not configured, keep the current behavior.
+
+
+## Motivation
+
+The default value of `--image-gc-high-threshold` option is 85%, and the default value of `imagefs.available` option is
+less than 15%, this will result in image garbage collection not taking effect until node gets disk pressure.
+
+There is no standard to judge whether the value of `--image-gc-high-threshold` is reasonable in different scenarios
+
+### Goals
+
+Discuss reasonable values of `--image-gc-high-threshold` for different scenarios, and constrain them by some means.
+Eventually protect users from inopportune configurations, and fix the defaults of `--image-gc-high-threshold` and `imagefs.available` to make they more sense.
+
+
+### Non-Goals
+
+
+## Proposal
+
+
+
+### User Stories (Optional)
+
+
+#### Story 1
+
+In big data computing scenarios, user often run some computing tasks . 
+When these tasks are completed, a large number of images are stored on the node. 
+At this time, user want to perform image garbage collection before the node disk pressure occurs. 
+There should be validation to protect that expectation
+
+#### Story 2
+
+Cluster administrator misconfigures `--image-gc-high-threshold` and `imagefs.available`.
+
+
+### Notes/Constraints/Caveats (Optional)
+
+
+### Risks and Mitigations
+
+
+## Design Details
+
+Add `ImageGCBeforeStorageEviction` feature gate to kubelet. 
+When the feature  is turned on, the value of `--image-gc-high-threshold` must be smaller than  value of `100 - imagefs.available`. 
```

Comment:
If the value of `--image-gc-high-threshold` is not less than "100 - imagefs.available", what will kubelet do? Will it panic, or will this field become ineffective? Can you explain this in detail here?